### PR TITLE
new readme changes

### PR DESCRIPTION
**Pull request by wduraes**
Added new readme files and updated main readme under IoT.

### CODE REVIEW

File: sdk/iot/hub/readme.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Azure IoT Clients for Embedded C: Azure IoT Hub Client
+
+Official Embedded C client for Azure IoT Hub.
+
+## Getting started
+
+- **Docs**: If you're new to the Azure SDK for Embedded C take a look at the [State Machine diagram](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/iot/doc/mqtt_state_machine.md#device-provisioning-service) that explains the high-level architecture, SDK components, and a clear view of SDK x Application responsibilities. 
+- **Samples**: [This page](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/iot/hub/samples) explains samples for the Azure SDK for Embedded C and how to use them.
+
+[TODO] update wiht the list of samples Hub, CDD, telemetry, method and twin
```
Comment (ericwol-msft) [inline sdk/iot/hub/readme.md:10]: wiht?

Comment (danewalton) [reply sdk/iot/hub/readme.md:10]: Yea I think I had the same comment on wellington's PR. Its probably best to merge his first and then this one.

File: sdk/iot/provisioning/readme.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+[opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+
+### Additional Helpful Links for Contributors
+
+Many people all over the world have helped make this project better.  You'll want to check out:
+
+* [What are some good first issues for new contributors to the repo?](https://github.com/azure/azure-sdk-for-c/issues?q=is%3Aopen+is%3Aissue+label%3A%22up+for+grabs%22)
+* [How to build and test your change](CONTRIBUTING.md#developer-guide)
+* [How you can make a change happen!](CONTRIBUTING.md#pull-requests)
+* Frequently Asked Questions (FAQ) and Conceptual Topics in the detailed [Azure SDK for C wiki][azure_sdk_for_c_wiki].
```
Comment (ericwol-msft) [inline sdk/iot/provisioning/readme.md:42]: links?

File: sdk/iot/provisioning/readme.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+# Azure IoT Clients for Embedded C: Azure IoT Device Provisioning Client
+
+Official Embedded C client for Azure IoT DPS - Device Provisioning Services.
+
+For more details about DPS check [Azure documentation](https://docs.microsoft.com/azure/iot-dps/)
+
+## Getting started
+
+- Pelase refer to the state machine diagram which describes the Azure SDK for Embedded C flow for DPS [here](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/iot/doc/mqtt_state_machine.md#device-provisioning-service).
```
Comment (ericwol-msft) [inline sdk/iot/provisioning/readme.md:9]: Avoid using DPS...long history. You can use "Provisioning Service"

File: sdk/iot/hub/readme.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+[opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+
+### Additional Helpful Links for Contributors
+
+Many people all over the world have helped make this project better.  You'll want to check out:
+
+* [What are some good first issues for new contributors to the repo?](https://github.com/azure/azure-sdk-for-c/issues?q=is%3Aopen+is%3Aissue+label%3A%22up+for+grabs%22)
+* [How to build and test your change](CONTRIBUTING.md#developer-guide)
+* [How you can make a change happen!](CONTRIBUTING.md#pull-requests)
+* Frequently Asked Questions (FAQ) and Conceptual Topics in the detailed [Azure SDK for C wiki][azure_sdk_for_c_wiki].
```
Comment (ericwol-msft) [inline sdk/iot/hub/readme.md:41]: Link?

File: sdk/iot/hub/readme.md (hunk 1 above)
Comment (wduraes) [reply sdk/iot/hub/readme.md]: Removed this TODO line from the page

File: README.md [hunk 5]
```diff
@@ -88,7 +88,7 @@ When we make an official release, we will create a unique git tag containing the
 
 ## SDK Architecture
 
-At the heart of our SDK is, what we refer to as, [Azure Core](sdk/core/core). This code defines several data types and functions for use by the client libraries that build on top of us such as an `Azure Storage Blob` client library and `IoT` client libraries. Here are some of the features that customers use directly:
+At the heart of our SDK is, what we refer to as, [Azure Core](sdk/core/core). This code defines several data types and functions for use by the client libraries that build on top of us such as an `Azure Storage Blob` client library and `IoT` [ client libraries](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/iot). Here are some of the features that customers use directly:
```
Comment (danewalton) [inline README.md:91]: I still think these should be links only:
```suggestion
At the heart of our SDK is, what we refer to as, [Azure Core](sdk/core/core). This code defines several data types and functions for use by the client libraries that build on top of us such as an [Azure Storage Blob](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/storage) client library and [IoT client libraries](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/iot). Here are some of the features that customers use directly:

```

File: sdk/iot/README.md [hunk 6]
```diff
@@ -1,14 +1,38 @@
...
+ [Receive cloud-to-device messages](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-c2d) | :heavy_check_mark: | Receive cloud-to-device messages and associated properties from IoT Hub.   
+ [Device Twins](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-device-twins) | :heavy_check_mark: | IoT Hub persists a device twin for each device that you connect to IoT Hub.  The device can perform operations like get twin document, subscribe to desired property updates.
+ [Direct Methods](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-direct-methods) | :heavy_check_mark: | IoT Hub gives you the ability to invoke direct methods on devices from the cloud.  
+ [DPS - Device Provisioning Service](https://docs.microsoft.com/azure/iot-dps/) | [Supported] | This SDK supports connecting your device to the Device Provisioning Service via, for example, [individual enrollment](https://docs.microsoft.com/azure/iot-dps/concepts-service#enrollment) using an [X.509 leaf certificate](https://docs.microsoft.com/azure/iot-dps/concepts-security#leaf-certificate).
```
Comment (danewalton) [inline sdk/iot/README.md:29]:
```suggestion
 [DPS - Device Provisioning Service](https://docs.microsoft.com/azure/iot-dps/) | :heavy_check_mark: | This SDK supports connecting your device to the Device Provisioning Service via, for example, [individual enrollment](https://docs.microsoft.com/azure/iot-dps/concepts-service#enrollment) using an [X.509 leaf certificate](https://docs.microsoft.com/azure/iot-dps/concepts-security#leaf-certificate).  

```

File: sdk/iot/README.md [hunk 7]
```diff
@@ -128,12 +152,37 @@ void my_telemetry_func()
...
+This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
+For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact
+[opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+
+### Additional Helpful Links for Contributors
+
+Many people all over the world have helped make this project better.  You'll want to check out:
+
+* [What are some good first issues for new contributors to the repo?](https://github.com/azure/azure-sdk-for-c/issues?q=is%3Aopen+is%3Aissue+label%3A%22up+for+grabs%22)
+* [How to build and test your change](CONTRIBUTING.md#developer-guide)
```
Comment (danewalton) [inline sdk/iot/README.md:174]: this `404`'s

File: sdk/iot/README.md [hunk 8]
```diff
@@ -128,12 +152,37 @@ void my_telemetry_func()
...
+* [How you can make a change happen!](CONTRIBUTING.md#pull-requests)
```
Comment (danewalton) [inline sdk/iot/README.md:175]: `404`'s

File: sdk/iot/README.md [hunk 9]
```diff
@@ -128,12 +152,37 @@ void my_telemetry_func()
...
+* Frequently Asked Questions (FAQ) and Conceptual Topics in the detailed [Azure SDK for C wiki][azure_sdk_for_c_wiki].
```
Comment (danewalton) [inline sdk/iot/README.md:176]: no link here

File: sdk/iot/hub/readme.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
+For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact
+[opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+
+### Additional Helpful Links for Contributors
+
+Many people all over the world have helped make this project better.  You'll want to check out:
+
+* [What are some good first issues for new contributors to the repo?](https://github.com/azure/azure-sdk-for-c/issues?q=is%3Aopen+is%3Aissue+label%3A%22up+for+grabs%22)
+* [How to build and test your change](CONTRIBUTING.md#developer-guide)
```
Comment (danewalton) [inline sdk/iot/hub/readme.md:37]: make sure to update these ones too

File: sdk/iot/provisioning/readme.md (hunk 3 above)
Comment (danewalton) [inline sdk/iot/provisioning/readme.md:9]:
```suggestion
- Please refer to the state machine diagram which describes the Azure SDK for Embedded C flow for DPS [here](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/iot/doc/mqtt_state_machine.md#device-provisioning-service). 
```

File: sdk/iot/provisioning/readme.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
+For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact
+[opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+
+### Additional Helpful Links for Contributors
+
+Many people all over the world have helped make this project better.  You'll want to check out:
+
+* [What are some good first issues for new contributors to the repo?](https://github.com/azure/azure-sdk-for-c/issues?q=is%3Aopen+is%3Aissue+label%3A%22up+for+grabs%22)
+* [How to build and test your change](CONTRIBUTING.md#developer-guide)
```
Comment (danewalton) [inline sdk/iot/provisioning/readme.md:40]: these too

File: sdk/iot/provisioning/readme.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+* Chat with other community members [![Join the chat at https://gitter.im/azure/azure-sdk-for-c](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/azure/azure-sdk-for-c?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
+
+### Reporting security issues and security bugs
+
+Security issues and bugs should be reported privately, via email, to the Microsoft Security Response Center (MSRC) <secure@microsoft.com>. You should receive a response within 24 hours. If for some reason you do not, please follow up via email to ensure we received your original message. Further information, including the MSRC PGP key, can be found in the [Security TechCenter](https://www.microsoft.com/msrc/faqs-report-an-issue).
+
+### License
+
+Azure SDK for Embedded C is licensed under the [MIT](LICENSE) license.
```
Comment (danewalton) [inline sdk/iot/provisioning/readme.md:54]: This `404`'s on every README. We should probably update all of these in another PR.

File: sdk/iot/README.md [hunk 13]
```diff
@@ -128,12 +153,37 @@ void my_telemetry_func()
 
 ## Contributing
 
-If you'd like to contribute to this library, please read the [contributing guide][azure_sdk_for_c_contributing] to learn more about how to build and test the code.
+For details on contributing to this repository, see the [contributing guide](https://github.com/Azure/azure-sdk-for-c/blob/master/CONTRIBUTING.md).
```
Comment (gilbertw) [inline sdk/iot/README.md:156]: This didn't merge with master correctly.  The contributing and License section should be the following

```
## Contributing

If you'd like to contribute to this library, please read the [contributing guide][azure_sdk_for_c_contributing] to learn more about how to build and test the code.

### License

Azure SDK for Embedded C is licensed under the [MIT][azure_sdk_for_c_license] license.

<!-- LINKS -->
[azure_sdk_for_c_contributing]: https://github.com/Azure/azure-sdk-for-c/blob/master/CONTRIBUTING.md
[azure_sdk_for_c_license]: https://github.com/Azure/azure-sdk-for-c/blob/master/LICENSE
```

File: sdk/iot/README.md [hunk 14]
```diff
@@ -1,14 +1,39 @@
-# Azure IoT Clients for Embedded C
+# Embedded C for Azure IoT Clients
```
Comment (gilbertw) [inline sdk/iot/README.md:1]: @ericwol-msft This new name pattern is different from how we name clients.  I would have expected
   Azure IoT Clients for Embedded C
ex:
https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/storage
https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/storage
https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/storage
https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/storage/azure-storage

https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/keyvault
https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/keyvault
https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/keyvault
https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-keyvault

File: README.md [hunk 15]
```diff
@@ -88,7 +88,8 @@ When we make an official release, we will create a unique git tag containing the
 
 ## SDK Architecture
 
-At the heart of our SDK is, what we refer to as, [Azure Core](sdk/core/core). This code defines several data types and functions for use by the client libraries that build on top of us such as an `Azure Storage Blob` client library and `IoT` client libraries. Here are some of the features that customers use directly:
+At the heart of our SDK is, what we refer to as, [Azure Core](sdk/core/core). This code defines several data types and functions for use by the client libraries that build on top of us such as an [Azure Storage Blob](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/storage) client library and [IoT client libraries](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/iot). Here are some of the features that customers use directly:
```
Comment (gilbertw) [inline README.md:91]:
```suggestion
At the heart of our SDK is, what we refer to as, [Azure Core](sdk/core/core). This code defines several data types and functions for use by the client libraries that build on top of us such as an [Azure Storage Blob](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/storage) client library and [Azure IoT client libraries](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/iot). Here are some of the features that customers use directly:

```

File: sdk/iot/provisioning/readme.md (hunk 2 above)
Comment (wduraes) [reply sdk/iot/provisioning/readme.md]: fixed.

File: sdk/iot/provisioning/readme.md (hunk 3 above)
Comment (wduraes) [reply sdk/iot/provisioning/readme.md]: fixed.

File: sdk/iot/hub/readme.md (hunk 4 above)
Comment (wduraes) [reply sdk/iot/hub/readme.md]: fixed.

File: sdk/iot/README.md (hunk 7 above)
Comment (wduraes) [reply sdk/iot/README.md]: fixed.

File: sdk/iot/README.md (hunk 8 above)
Comment (wduraes) [reply sdk/iot/README.md]: fixed.

File: sdk/iot/README.md (hunk 9 above)
Comment (wduraes) [reply sdk/iot/README.md]: fixed.

File: sdk/iot/hub/readme.md (hunk 10 above)
Comment (wduraes) [reply sdk/iot/hub/readme.md]: done!

File: sdk/iot/provisioning/readme.md (hunk 11 above)
Comment (wduraes) [reply sdk/iot/provisioning/readme.md]: done too.

File: sdk/iot/provisioning/readme.md (hunk 12 above)
Comment (wduraes) [reply sdk/iot/provisioning/readme.md]: fixed. not sure what happened here...

File: sdk/iot/README.md (hunk 13 above)
Comment (wduraes) [reply sdk/iot/README.md]: updated!

File: README.md (hunk 15 above)
Comment (wduraes) [reply README.md]: done!

File: sdk/iot/provisioning/readme.md [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+
+- **Samples**: [This page](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/iot/provisioning/samples) explains samples for the Azure SDK for Embedded C IoT Provisioning Client and how to use them.
+
+## Need help?
+
+* File an issue via [Github Issues](https://github.com/Azure/azure-sdk-for-c/issues/new/choose).
+* Check [previous questions](https://stackoverflow.com/questions/tagged/azure+c) or ask new ones on StackOverflow using
+  the `azure` and `c` tags.
+
+## Contributing
```
Comment (gilbertw) [inline sdk/iot/provisioning/readme.md:19]: The contributing section should be a reference to the repo contributing section.  That way we only need to update one location.

```
## Contributing

If you'd like to contribute to this library, please read the [contributing guide][azure_sdk_for_c_contributing] to learn more about how to build and test the code.

### License

Azure SDK for Embedded C is licensed under the [MIT][azure_sdk_for_c_license] license.

<!-- LINKS -->
[azure_sdk_for_c_contributing]: https://github.com/Azure/azure-sdk-for-c/blob/master/CONTRIBUTING.md
[azure_sdk_for_c_license]: https://github.com/Azure/azure-sdk-for-c/blob/master/LICENSE
```

File: sdk/iot/hub/readme.md [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+- **Docs**: If you're new to the Azure SDK for Embedded C take a look at the [State Machine diagram](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/iot/doc/mqtt_state_machine.md#device-provisioning-service) that explains the high-level architecture, SDK components, and a clear view of SDK x Application responsibilities. 
+- **Samples**: [This page](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/iot/hub/samples) explains samples for the Azure SDK for Embedded C and how to use them.
+
+## Need help?
+
+* File an issue via [Github Issues](https://github.com/Azure/azure-sdk-for-c/issues/new/choose).
+* Check [previous questions](https://stackoverflow.com/questions/tagged/azure+c) or ask new ones on StackOverflow using
+  the `azure` and `c` tags.
+
+## Contributing
```
Comment (gilbertw) [inline sdk/iot/hub/readme.md:16]: Contributing and license should refer to the repo contributing/license

```
## Contributing

If you'd like to contribute to this library, please read the [contributing guide][azure_sdk_for_c_contributing] to learn more about how to build and test the code.

### License

Azure SDK for Embedded C is licensed under the [MIT][azure_sdk_for_c_license] license.

<!-- LINKS -->
[azure_sdk_for_c_contributing]: https://github.com/Azure/azure-sdk-for-c/blob/master/CONTRIBUTING.md
[azure_sdk_for_c_license]: https://github.com/Azure/azure-sdk-for-c/blob/master/LICENSE
```

File: sdk/iot/README.md [hunk 18]
```diff
@@ -130,10 +154,27 @@ void my_telemetry_func()
```
Comment (gilbertw) [inline sdk/iot/README.md:154]: Contributing section should refer to repo contributing section

```
## Contributing

If you'd like to contribute to this library, please read the [contributing guide][azure_sdk_for_c_contributing] to learn more about how to build and test the code.

### License

Azure SDK for Embedded C is licensed under the [MIT][azure_sdk_for_c_license] license.

<!-- LINKS -->
[azure_sdk_for_c_contributing]: https://github.com/Azure/azure-sdk-for-c/blob/master/CONTRIBUTING.md
[azure_sdk_for_c_license]: https://github.com/Azure/azure-sdk-for-c/blob/master/LICENSE
```

**Comment by gilbertw** [comment]
The contributing and license sections in all Readme files were consolidated to reference the root contributing/license.  Please point to that file so that we can update in one locaation.  If something helpful to contributing is missing please add to the root contributing doc such that all readme's benefit.

File: sdk/iot/README.md (hunk 18 above)
Comment (wduraes) [reply sdk/iot/README.md]: done!

File: sdk/iot/provisioning/readme.md (hunk 16 above)
Comment (wduraes) [reply sdk/iot/provisioning/readme.md:19]: done!

File: sdk/iot/hub/readme.md (hunk 17 above)
Comment (wduraes) [reply sdk/iot/hub/readme.md:16]: done!

**Comment by wduraes** [comment]
work completed for this PR. all comments and suggestion have been addressed for this initial review of readme files. we'll keep reviewing them as we get feedback and questions from customers and partners.

**Pull request merged**